### DISCUSSION TITLE
Get the centroids of the voxel cells

### DISCUSSION THREAD

**Discussion by SpatialDigger** [pyvista/pyvista#3742] [Q&A] [answered]
Using this tutorial as a convenient dataset:
https://banesullivan.com/pyvista/examples/voxelize.html

Imagine I had 'lost' the original points, how would I get or recalculate the centroids?

Currently the grid.points returns the corner nodes of the cells

grid,center returns the center of the entire grid, I want the centers of each cell.

1st solution:
This seems rather slow, but it does the job, I'm open to improved solutions.

```
    def get_centroids(self):
        vox = self.voxels['group']
        i=0
        centroids = []
        while i < vox.n_cells:
            cell = vox.extract_cells(i)
            centroids.append(cell.center)
            i += 1
        return np.asarray(centroids)
```

**Comment by banesullivan** [comment, marked as answer]
The `cell_centers()` filter will compute the voxel cell centers: https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSet.cell_centers.html#pyvista.DataSet.cell_centers
